recipes/delphes/meta.yaml

```diff
@@ -0,0 +1,75 @@
+{% set name = "delphes" %}
+{% set version = "3.5.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/delphes/{{ name }}/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 30a2536e0c8f47d633ecf3b4d51ec9924af54538e492470ad47b4ed34342f063
+  patches:
+    # Fixes builds for macOS
+    # c.f. https://github.com/delphes/delphes/pull/128
+    - Delphes-ClosestPair2D-fwdDeclaration.patch
+    # c.f. https://github.com/delphes/delphes/commit/64de5a479b94beba357e73c926cdc6ee9114ceec
+    - remove_binary_function.patch
+
+build:
+  skip: true  # [win]
+  script:
+    # Make the examples location independent by removing the 'external/' prefix
+    # in includes to instead use the installed versions
+    - grep -r -l "external/" examples/ | xargs sed -i 's|external/||g'  # [linux]
+    - grep -r -l "external/" examples/ | xargs sed -i '' 's|external/||g'  # [osx]
+
+    - CXXFLAGS="${CXXFLAGS}" cmake -DCMAKE_INSTALL_PREFIX=$PREFIX -DPYTHIA8_DATA=$(pythia8-config --datadir) -S . -B build
+    - cmake build -LH
+    - cmake --build build --clean-first --parallel="${CPU_COUNT}"
+    - cmake --install build
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ stdlib('c') }}
+    - cmake
+    - make
+  host:
+    - root_base >=6.28.0
+    - pythia8
+  run:
+    - root_base >=6.28.0
+    - pythia8
```

Comment:
Apparently from https://github.com/conda-forge/staged-recipes/pull/27713/files#r1798235502 the `run` dependencies for C++ projects should not exist and should be controlled through `run_exports`? https://conda-forge.org/docs/maintainer/pinning_deps/#specifying-run_exports

I don't know how that works though, so I'll have to read up on that.

Reply:
From https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#export-runtime-requirements I'm not really sure how to do this though. We don't want to be overly restrictive on the versions of `root` and `pythia8` as libraries, so I don't know which https://docs.conda.io/projects/conda-build/en/stable/resources/variants.html#extra-jinja2-functions to use here. I also don't understand why some example I see are using `run_exports` on the package build itself.

@chrisburr As you've done this for both `pythia8` and `root`/`root_base` input here is welcome.

Reply:
`run_exports` should be used on packages that provide a shared library that you expect people to link against. You generally only provide run_exports for the package you're currently building (the more general case is more of an expert feature that you're unlikely to need).

> We don't want to be overly restrictive on the versions of root and pythia8 as libraries,

You have no control over this, both have bad ABI stability policies so you're stuck to the version you built with (slightly more relaxed for Pythia IIRC).

Reply:
```suggestion
```

Reply:
I also don't understand why removing the _lower bound_ on `root_base`

```diff
  host:
-    - root_base >=6.28.0
+    - root_base
```

is now causing build errors

```
Could not solve for environment specs
The following packages are incompatible
├─ gxx_linux-64 13.*  is requested and can be installed;
└─ root_base >=6.32.0,<6.32.1.0a0  is not installable because it requires
   └─ gxx_linux-64 12.* , which conflicts with any installable versions previously reported.
```

`root_base` (https://github.com/conda-forge/root-feedstock) hasn't been rerendered or released in multiple weeks, so I don't understand why this build error now appears when I previously only had a lower bound. :?

Reply:
Ah. Okay given https://conda-forge.org/docs/maintainer/pinning_deps/#globally-pinned-packages and https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/0b6ce4dc053ac4b92814f7004e0dae13755c0cb3/recipe/conda_build_config.yaml#L823-L825

```yaml
...
root_base:
  - 6.30.4
  - 6.32.0
...
```

I guess so. So to be able to build Delphes we need to first do a migration on `root_base` in https://github.com/conda-forge/conda-forge-pinning-feedstock?

Reply:
The reason I assume we need to update the global pin instead of rerendering the https://github.com/conda-forge/root-feedstock is that `root_base` is now on `v6.32.2` and so it would be easier to update the global pin than to do a rerender of an older version of the feedstock. I might be wrong(?).